src/react/components/GenericForm.jsx

```diff
@@ -18,112 +62,180 @@ const { TextArea } = Input;
  * @param {Array} props.fields        - array of objects containing all single fields
  *
  */
-export const GenericForm = ({ props }) => {
-    const {
-        formSettings,
-        fields
-    } = props;
 
-    const [form] = Form.useForm();
+export class GenericForm extends React.Component {
+    constructor (props) {
+        super(props);
+        
+        this.formRef = React.createRef();
```

Comment:
I'm not sure about using ref. Can you tell a bit more why we need to use it here?

Reply:
Removed ref since we don't need that anymore with `initialValue` prop.